docs/src/tutorial/pure.rst

```diff
@@ -399,6 +399,46 @@ efficient C code. In other cases, however, where the generated C code could
 benefit from these type hints but does not currently, help is welcome to
 improve the type analysis in Cython.
 
+Reference table
+^^^^^^^^^^^^^^^
+
+The following reference table documents how type annotations are
+interpreted. Cython 0.29 behaviour is only shown where it differs
+from Cython 3.0 behaviour.
+
++---------------------+--------------------+---------------------------+
```

Comment:
Might be better done as a "long table" (https://tables-with-sphinx.readthedocs.io/en/latest/csv-table.html) possibly? It won't take much more to make drawing the table too much of a pain